### PR TITLE
do not add a workflow to yourself if the sub-collections includes collection itself

### LINKED ISSUE #69: release-members step now fails on all collection objects
This is the error message: release-members : Failed to retrieve resource: put https://sul-lyberservices-prod.stanford.edu/workflow/dor/objects/druid:xs573nh8717/workflows/releaseWF/release-members?current-status=queued (HTTP status 500)

### PR DESCRIPTION
**Pull request by peetucket**
Also, refactor tests and be consistent in how item_members and sub_collections return an empty array if there are no items

*hopefully* fixes #69

### CODE REVIEW

File: lib/dor/item.rb [hunk 1]
```diff
@@ -43,7 +43,7 @@ def sub_collections
         @sub_collections += members['sets'] if members['sets']
         @sub_collections += members['collections'] if members['collections']
       end
-      @sub_collections
+      @sub_collections.delete_if {|collection| collection['druid'] == druid} # if this is a collection, do not include yourself in the sub-collection list
```
Comment (cbeer) [inline lib/dor/item.rb]: Shouldn't the `delete_if` statement live inside the `unless @sub_collections` block? (Otherwise, you're presumably unnecessarily running the `delete_if` every time `sub_collections` is called).

Comment (peetucket) [inline lib/dor/item.rb]: yes, good catch, will fix and squash the commit
